cakechat/dialog_model/model.py

```diff
@@ -583,8 +584,10 @@ def print_matrices_weights(self):
         total_network_size = 0
         for p, v in zip(params, values):
             param_size = float(v.nbytes) / 1024 / 1024
+            # Work around numpy/python 3 regression: 
+            # http://www.markhneedham.com/blog/2017/11/19/python-3-typeerror-unsupported-format-string-passed-to-numpy-ndarray-__format__/
             laconic_logger.info('\t{0:<40} dtype: {1:<10} shape: {2:<12} size: {3:<.2f}M'.format(
-                p.name, v.dtype, v.shape, param_size))
+                p.name, repr(v.dtype), repr(v.shape), param_size))
```

Comment:
maybe it's better just not to limit length of string representations of these params, so there would be no need in a workaround